lib/Logger/Logger.cpp

```diff
@@ -325,88 +325,278 @@ void Logger::log(char const* function, char const* file, int line,
   if (_cachedPid == 0) {
     _cachedPid = Thread::currentProcessId();
   }
-  TRI_ASSERT(_cachedPid != 0);
-  out.push_back('[');
-  StringUtils::itoa(uint64_t(_cachedPid), out);
 
-  if (_showThreadIdentifier) {
-    out.push_back('-');
-    StringUtils::itoa(uint64_t(Thread::currentThreadNumber()), out);
-  }
+  std::string out;
+  size_t offset = 0;
+
+  if (Logger::_useJson) {
+    // construct JSON output
+    StringBuffer buf;
+    buf.appendChar('{');
+
+    // current time
+    {
+      static char const s[] = "time";
 
-  // log thread name
-  if (_showThreadName) {
-    char const* threadName = Thread::currentThreadName();
-    if (threadName == nullptr) {
-      threadName = "main";
+      buf.appendChar('"');
+      buf.appendText(s, sizeof(s) - 1);
+      buf.appendText("\":", 2);
+
+      std::chrono::system_clock::time_point now = std::chrono::system_clock::now();
+      std::chrono::system_clock::time_point tp(std::chrono::duration_cast<std::chrono::seconds>(now.time_since_epoch()));
+      buf.appendInteger(uint64_t(std::chrono::duration_cast<std::chrono::seconds>(tp.time_since_epoch()).count()));
     }
 
-    out.push_back('-');
-    out.append(threadName);
-  }
+    // prefix
+    if (!_outputPrefix.empty()) {
+      static char const s[] = "prefix";
 
-  out.append("] ", 2);
+      buf.appendText(",\"", 2);
+      buf.appendText(s, sizeof(s) - 1);
+      buf.appendText("\":", 2);
+      buf.appendJsonEncoded(_outputPrefix.c_str(), _outputPrefix.size());
+    }
 
-  if (_showRole && _role != '\0') {
-    out.push_back(_role);
-    out.push_back(' ');
-  }
+    // pid
+    {
```

Comment:
While we are on it, we should also add an option that makes logging of the pid configurable.
When in Docker containers, logging the pid makes no sense, so we could as well add an option to turn it off if the user wants to opt out. This should be a new logging option, which should be handled by both logging implementations.
It should default to logging the pid.

Reply:
ok

Reply:
Added the option.